animation_nodes/nodes/gpencil/gp_object_output.py

```diff
@@ -0,0 +1,89 @@
+import bpy
+from bpy.props import *
+from ... base_types import AnimationNode, VectorizedSocket
+
+class GPObjectOutputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPObjectOutputNode"
+    bl_label = "GP Object Output"
+    errorHandlingType = "EXCEPTION"
+
+    useLayerList: VectorizedSocket.newProperty()
+
+    def create(self):
+        socket = self.newInput("Object", "Object", "object")
+        socket.defaultDrawType = "PROPERTY_ONLY"
+        socket.objectCreationType = "GPENCIL"
+        self.newInput(VectorizedSocket("GPLayer", "useLayerList",
+            ("Layer", "layer"), ("Layers", "layers")))
+        self.newOutput("Object", "Object", "object")
+
+    def getExecutionFunctionName(self):
+        if self.useLayerList:
+            return "execute_LayerList"
+        else:
+            return "execute_Layer"
+
+    def execute_Layer(self, object, layer):
+        if object is None: return None
+        gpencil = self.getObjectData(object)
+
+        if layer is None: return object
+        self.setLayerData(gpencil, layer)
+        gpencil.layers.active.frames.update()
+        return object
+
+    def execute_LayerList(self, object, layers):
+        if object is None: None
+        gpencil = self.getObjectData(object)
+
+        if len(layers) == 0: return object
+        for layer in layers:
+            if layer is None: return
+            self.setLayerData(gpencil, layer)
+        gpencil.layers.active.frames.update()
+        return object
+
+    def setLayerData(self, gpencil, layer):
+        gpencilLayer = self.getLayer(gpencil, layer)
+        gpencilLayer.clear()
+
+        for frame in layer.frames:
+            gpFrame = gpencilLayer.frames.new(frame.frameNumber, active = True)
+            gpFrame.clear()
+            for stroke in frame.strokes:
+                gpStroke = self.setStrokeProperties(gpFrame.strokes.new(), stroke)
```

Comment:
I think you should create the stroke in a separate statement. And don't return anything in the `setStrokeProperties` method.

Reply:
Changed.